src/coreclr/tests/src/JIT/Regression/JitBlue/Runtime_35144/Runtime_35144.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This test was the repro case for issue #35144.
+// Until interop is supported for vectors, it is difficult to validate
+// that the ABI is correctly implemented, but this test is here to enable
+// these cases to be manually verified (and diffed).
+//
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+
+#pragma warning disable 0169 // warning CS0169: The field '{0}' is never used
+struct WrappedVector64  { Vector64<byte> _; }
+struct WrappedVector128 { Vector128<byte> _; }
+struct WrappedVector256 { Vector256<byte> _; }
+
+// Was incorrectly treated as non-HVA: passed in x0, x1.
+// Should be recognized as HVA(SIMD8) and passed in d0, d1.
+struct S1 { WrappedVector64 x; Vector64<byte> y; }
+
+// Was incorrectly treated as HFA(double): passed in d0, d1.
+// Should be passed as non-HFA in x0, x1.
+struct S2 { WrappedVector64 x; double y; }
+
+// Incorrectly treated as HVA(simd16): passed in q0, q1, q2.
+// Should be passed by reference as non-HFA.
+struct S3 { Vector128<byte> x; WrappedVector256 y; }
+
+static class Runtime_35144
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static void Foo<T>(T x) { }
```

Comment:
Could you change this test to pass an object after the T parameter? I would like it for GC Stress to cover the case where we use the calling convention processing logic of the VM to parse all of this.

Reply:
In particular make the function be written as:

```
static void Foo<T>(T x, object o)
{
    if (o.GetType() != typeof(string)) throw new Exception();
    if (((string)o) != "SomeString") throw new Exception();
}
```
and call it passing a string "SomeString" as the extra parameter.

Reply:
Also, if you could mark the Foo<T> method as AggressiveOptimization so that it isn't compiled by crossgen that will also help chase down the case where crossgen compiles this different from what the normal runtime does.

Reply:
@davidwrighton - Thanks for the review, and for the suggestions for improving the test!